lua/auto-dark-mode/interval.lua

```diff
@@ -67,6 +67,33 @@ local function sync_theme(appearance)
 	end
 end
 
+-- Uses a subprocess to monitor the system for the current dark mode setting.
+-- The callback is called with the plaintext stdout response of the query.
+---@param callback? fun(stdout: string, stderr: string): nil
+---@return nil
+M.monitor_dark_mode = function (callback)
+    -- if no callback is provided, use a no-op
+    callback = callback or function() end
+
+    vim.fn.jobstart(M.state.monitor_command,
```

Comment:
I see in documentation that `vim.system()` is preferable over `vim.fn.jobstart()`. Could it be done in the same way as in other methods?

Reply:
No, as `vim.system` only provides the command output once the command has finished executing (which is obviously never in the case of the monitoring command). I checked the docs, but there doesn't seem to be the option to parse the output whilst the command is still executing.